### PR TITLE
Read from and write to the Firestore database so that users can persist information

### LINKED ISSUE #1: 1. As a component, I want to be able to read from and write to the Firestore database so that users can persist information.
This is 1 of 2 stories that should be completed in the first week to enable further development of the app.
      AC:
      *  [`firebase`](https://www.npmjs.com/package/firebase) has been added as a project dependency
      * Making a change in the Firestore database will update the data in the app
      * Making a change in the app will update the data in Firestore
      Notes:
      * Don't do too much here. In the past, pairs have built the whole add item form in the first week. The best approach is just to make a throw away view with a `<button/>` and a `<div/>`. Click the `<button/>` to send some data to the database. Have the data show up in the `<div/>`. Simple!
      * If you need additional direction on how to use Firebase, you can read through the [Quickstart Guide](https://firebase.google.com/docs/firestore/quickstart) for understanding how to [add items to Firestore](https://firebase.google.com/docs/firestore/quickstart#add_data), as well as the Snapshot Guide for [creating a snapshot subscription](https://firebase.google.com/docs/firestore/query-data/listen#listen_to_multiple_documents_in_a_collection) to read and update data in real-time.
      ---
      Quick Tips:
      <details>
        <summary>How to send data to Firestore</summary>
        ```javascript
      import { collection, addDoc } from 'firebase/firestore'
      try {
          const docRef = await addDoc(collection(db, COLLECTION_NAME), {
            property: 'value'
            // ...data
          })
          // Success!
          console.log(docRef.id)
      } catch (e) {
          // There was an error sending data to Firestore
          console.error(e)
      }
        ```
      </details>
      <details>
        <summary>How to listen to data from Firestore</summary>
        ```javascript
      import { collection, onSnapshot } from 'firebase/firestore'
      const [docs, setDocs] = useState([])
      useEffect(() => {
          const unsubscribe = onSnapshot(collection(db, COLLECTION_NAME), (snapshot) => {
            // Firestore requires that you loop through the snapshot to access the docs,
            // instead of just setting the snapshot as the value of the state
            const snapshotDocs = []
            snapshot.forEach(doc => snapshotDocs.push(doc))
            setDocs(snapshotDocs)
          })
          return () => {
            // Used to remove the snapshot listener when the component is unmounted
            unsubscribe();
          }
      }, [])
        ```
      </details>

### PR DESCRIPTION
**Pull request by vividavila98**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

In this PR, we created two new files: `firebase.js` and `List.js`. In `firebase.js` we added the config information to connect firebase and firestore to the app. In `List.js` is where we displayed the items of our database and sent new data to the database.

In `firebase.js` we found that using `import firebase from "firebase"` conflicts with `import { initializeApp } from 'firebase/app'` so we used `import { getFirestore } from 'firebase/firestore'`  to connect to the database instead. 

In `List.js` we learned that we have to manually add the document id when pushing the document data to an array. 

## Related Issue
closes #1 

## Acceptance Criteria

- [x] firebase has been added as a project dependency
- [X] Making a change in the Firestore database will update the data in the app
- [X] Making a change in the app will update the data in Firestore

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
| ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

Screen Shot 2022-01-12 at 4 06 04 PM

### After
Screen Shot 2022-01-12 at 4 08 31 PM

## Testing Steps / QA Criteria
1. npm install
2. npm start
3. You should expect to see the collection data in the UI on the site
4.  To send data to Firestore, click the "Send data to database" button.
5. UI is updated with the data that is hard coded into sendData()

### CODE REVIEW

**Review by sanrafa — APPROVED** [review]
I pulled this down and it seems to be working exactly as intended. The code is well-written and easy to follow even without Firebase experience. Great work!

**Review by devaratpatel — COMMENTED** [review]
PR looks good for merge! Good work!

File: src/firebase.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { getFirestore } from 'firebase/firestore';
+
+const firebaseConfig = {
+  apiKey: process.env.API_KEY,
+  authDomain: 'tcl-37-smart-shopping-list.firebaseapp.com',
+  projectId: 'tcl-37-smart-shopping-list',
+  storageBucket: 'tcl-37-smart-shopping-list.appspot.com',
+  messagingSenderId: '867331867909',
+  appId: '1:867331867909:web:c5268888f54fa923ec236c',
+};
```
Comment (yenly) [inline src/firebase.js:11]: 👏🏽 Well done in using environment variables!

**Review by yenly — APPROVED** [review]
Fantastic write up in your PR! I appreciated the screen captures and the testing steps. It made reviewing this set of work quite easy.

File: pull_request_template.md [hunk 2]
```diff
@@ -18,26 +18,25 @@ _For an example of how to fill this template out, [see this Pull Request](https:
...
 ### After
 
 <!-- If UI feature, take provide screenshots -->
 
-
 ## Testing Steps / QA Criteria
 
 <!-- Provide steps the other cohort members and mentors need to follow to properly test your additions. -->
 
+<!-- Test comment, creating draft PR -->
```
Comment (tannaurus) [inline pull_request_template.md]: What is the purpose of these changes?

File: src/List.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from 'react';
+import { addDoc } from 'firebase/firestore';
+import db from './firebase';
+import { collection, onSnapshot } from 'firebase/firestore';
+
+export default function Db() {
+  const [docs, setDocs] = useState([]);
+  useEffect(() => {
```
Comment (tannaurus) [inline src/List.js:8]: I have some thoughts around how we're interacting with firebase here. I see this is approach that's been recommended via the issue itself.. imo this is the ["JavaScript" way](https://firebase.google.com/docs/firestore/query-data/listen) of doing it where [there is a more "React" way](https://github.com/CSFrequency/react-firebase-hooks/tree/v4.0.x/firestore#usecollection) of communicating with Firebase that I think will be better suited for our app. With this approach, we're using the standard firebase JS library and effectively writing our own hook every time we wish to interact with firebase in this way. We can skip this step by taking advantage of a[ common Firebase library](https://github.com/CSFrequency/react-firebase-hooks/tree/v4.0.x/firestore) that is written with React developers in mind. It exposes a function that does what we're doing here with a bit less hassle. You can read up on it [here](https://github.com/CSFrequency/react-firebase-hooks/tree/v4.0.x/firestore#usecollection)

File: src/firebase.js (hunk 1 above)
Comment (tannaurus) [reply src/firebase.js:11]: Love this too! We'll likely have to configure GitHub to have a copy of this api key so [our previews](https://github.com/the-collab-lab/tcl-37-smart-shopping-list/pull/17#issuecomment-1009238597) still work

**Review by tannaurus — APPROVED** [review]

File: pull_request_template.md (hunk 2 above)
Comment (danainjax) [reply pull_request_template.md]: Thanks for catching this. It will be removed. Originally placed to test the commit.

File: src/List.js (hunk 3 above)
Comment (danainjax) [reply src/List.js:8]: Thank you for this feedback! Checking it out.

Comment (vividavila98) [reply src/List.js:8]: @tannaurus Thank you for sharing this, that makes a lot of sense!

@danainjax Since issue 4 deals with submitting to the database, I can make this change with Dev if you don't mind. Let me know your thoughts!

Comment (danainjax) [reply src/List.js:8]: Sounds great, Viviana. I tried to wrap my brain around it this morning and got stuck. Hoping I can make office hours this week and talk through it. Feel free to proceed. 👍🏻
